### ISSUE TITLE
rerender for each instance of use-sound

### ISSUE THREAD

**Issue by martinjuhasz** [joshwcomeau/use-sound#42]
When using multiple use-sound instances, the page re-renders its content depending on how many hooks are used.

F.e.:
```
const Page = () => {
  const sound1 = useSound(sample1)
  const sound2 = useSound(sample2)
  const sound3 = useSound(sample3)

  console.log('render')
  return (<>hello</>)
}
```

will log `render` like 14 times. See this [codesandbox](https://codesandbox.io/s/wispy-sound-rmybc).

Is this the expected behaviour?

**Comment by andrewmclagan** [comment]
Also seeing this

**Comment by yomed** [comment]
I dug into this a little bit and found that additional rerenders upon calling `play()` are caused by this internal use of state:
https://github.com/joshwcomeau/use-sound/blob/d920ebb70c5d42c9ad74080bd6a0737b07b59b3b/src/index.ts#L116-L127

**Comment by tillschomborg** [comment]
I also recognized this and I have additionally a known react issue but I don't know how to fix it in this case.

This is my code:

`import useSound from 'use-sound';
import Sounds from '../../sounds';
[...]

const BoopButton = () => {
  let filename = Sounds[_AnotherStateVariable_].path; //the state of this parameter changes on user input and it is also a state variable
  const [play] = useSound(filename);

  return return <button onClick={() => { play(); doSomethingElse(); }}>Boop</button>;
};
const doSomethingElse = () => {
  [...];
  setAnotherStateVariable('default');
}
`

The Warning:
Warning: Can't perform a React state update on an unmounted component. This is a no-op, but it indicates a memory leak in your application. To fix, cancel all subscriptions and asynchronous tasks in a useEffect cleanup function.
    in BoopButton

**Comment by aelainelong** [comment]
Any updates on this?

**Comment by Kaiyiwing** [comment]
I have same Warning, even after I update use-sound to 2.0.1

**Comment by KarandikarMihir** [comment]
Any update on this? The hook is causing seemingly unnecessary rerenders :(

**Comment by joshwcomeau** [comment]
I don't have the bandwidth to look into this right now, but you can avoid having multiple `useSound` calls by using a sprite. See this example: https://github.com/joshwcomeau/use-sound/blob/master/stories/demos/DrumMachine.js#L28

**Comment by binajmen** [comment]
> Any update on this? The hook is causing seemingly unnecessary rerenders :(

Same here, a lot of unnecessary rerenders while handling a large amount of items. Did you find a fix?

**Comment by yomed** [comment]
Not a fix, but a workaround would be to use https://github.com/ds300/patch-package and comment out the lines I mentioned above: https://github.com/joshwcomeau/use-sound/issues/42#issuecomment-735530024 (only if you aren't using that functionality.

**Comment by joshwcomeau** [comment]
Ok, so I just published 4.0.0. The number of renders should be quite a bit lower now. In addition to removing the lines that @yomed found, I've made a few other tweaks that should help.

In order to lower the render count, I had to ditch the `isPlaying` boolean. This is easy enough to implement on your own, if you need it. A new Storybook story, `ShowWhilePlaying`, shows an example.

When loading a component with multiple hooks, it will still re-render multiple times per hook. This is because sounds load at different times, and once the sound is loaded, we need to set it into state so that it can be interacted with.

If you have multiple hooks per component, the recommendation is still to use a sprite, as I mentioned above. This is a good idea even without considering the render count.
